--- gitlab-runner/shells/abstract.go~	2019-10-28 14:11:26.000000000 +0200
+++ gitlab-runner/shells/abstract.go	2019-11-10 02:47:02.568781451 +0200
@@ -337,7 +337,11 @@
 
 func (b *AbstractShell) writeCheckoutCmd(w ShellWriter, build *common.Build) {
 	w.Notice("Checking out %s as %s...", build.GitInfo.Sha[0:8], build.GitInfo.Ref)
-	w.Command("git", "checkout", "-f", "-q", build.GitInfo.Sha)
+	if build.GitInfo.RefType == common.RefTypeTag {
+		w.Command("git", "checkout", "-f", "-q", build.GitInfo.Sha)
+	} else {
+		w.Command("git", "checkout", "-fq", "-B", build.GitInfo.Ref, build.GitInfo.Sha)
+	}
 
 	cleanFlags := build.GetGitCleanFlags()
 	if len(cleanFlags) > 0 {
