--- gitlab-ci-multi-runner-1.7.0/src/gitlab.com/gitlab-org/gitlab-ci-multi-runner/shells/abstract.go~	2016-10-21 23:19:09.000000000 +0300
+++ gitlab-ci-multi-runner-1.7.0/src/gitlab.com/gitlab-org/gitlab-ci-multi-runner/shells/abstract.go	2016-10-24 11:51:14.842526005 +0300
@@ -99,7 +99,7 @@
 
 func (b *AbstractShell) writeCheckoutCmd(w ShellWriter, build *common.Build) {
 	w.Notice("Checking out %s as %s...", build.Sha[0:8], build.RefName)
-	w.Command("git", "checkout", "-f", "-q", build.Sha)
+	w.Command("git", "checkout", "-fq", "-B", build.RefName, build.Sha)
 }
 
 func (b *AbstractShell) cacheFile(build *common.Build, userKey string) (key, file string) {
