Last-Update: 2016-08-24
Forwarded: not-needed
Bug-Upstream: https://gitlab.com/gitlab-org/gitlab-ci-multi-runner/issues/1504
Author: Dmitry Smirnov <onlyjob@debian.org>
Description: do not incorporate Docker image into executable.

--- gitlab-ci-multi-runner/executors/docker/executor_docker.go~	2017-05-02 13:12:18.000000000 +0300
+++ gitlab-ci-multi-runner/executors/docker/executor_docker.go	2017-05-16 08:08:28.042038918 +0300
@@ -228,16 +228,11 @@
 		return &image, nil
 	}
 
-	data, err := Asset("prebuilt-" + architecture + prebuiltImageExtension)
-	if err != nil {
-		return nil, fmt.Errorf("Unsupported architecture: %s: %q", architecture, err.Error())
-	}
-
 	s.Debugln("Loading prebuilt image...")
 
 	ref := prebuiltImageName
 	source := types.ImageImportSource{
-		Source:     bytes.NewBuffer(data),
+		Source:      "/var/lib/gitlab-runner/prebuilt-" + architecture + ".tar.xz",
 		SourceName: "-",
 	}
 	options := types.ImageImportOptions{
